src/query.ts

```diff
@@ -55,6 +63,9 @@ export function parseQuery(search: string): LocationQuery {
     // allow the = character
     let eqPos = searchParam.indexOf('=')
     let key = decode(eqPos < 0 ? searchParam : searchParam.slice(0, eqPos))
+    if (!isAllowedQueryKey(key)) {
```

Comment:
Wouldn't this suffice to avoid setting invalid keys

```suggestion
    if (Object.prototype.hasOwnProperty(key)) {
```

`prototype` is a valid key BTW

Reply:
I have abstracted logic into a separate function for easier readability and scaling.

Reply:
I'm saying that `Object.prototype.hasOwnProperty()` is the only check needed

Reply:
@posva I simplified condition. But I propose to keep it in a separate function `isAllowedQueryKey`.

Reply:
Thanks, I think the extra function here adds no clarity but a little comment above saying this ignores `?__proto__&toString` is enough